gswikichat/vector_store_interface.py

```diff
@@ -57,112 +60,55 @@
-# sentence_transformer_model = 'jinaai/jina-embeddings-v2-base-de'
-# Cannot find or load the embedding model
-# Unknown minutes to batch 82
-
-# https://huggingface.co/aari1995/German_Semantic_STS_V2
-# sentence_transformer_model = 'aari1995/German_Semantic_STS_V2'
-# 75 minutes to batch 82
-
-# https://huggingface.co/Sahajtomar/German-semantic
-# sentence_transformer_model = 'Sahajtomar/German-semantic'
-# 72 minutes to batch 82
 
 # https://huggingface.co/svalabs/german-gpl-adapted-covid
 sentence_transformer_model = 'svalabs/german-gpl-adapted-covid'
-# 2 minutes to batch 82
-
-# https://huggingface.co/PM-AI/bi-encoder_msmarco_bert-base_german
-# sentence_transformer_model = 'PM-AI/bi-encoder_msmarco_bert-base_german'
-# 14 minutes to batch 82
-
-# https://huggingface.co/JoBeer/german-semantic-base
-# sentence_transformer_model = 'JoBeer/german-semantic-base'
-# 22 minutes to batch 82
-
-print(f'Sentence Transformer Name:{sentence_transformer_model}')
+print(f'Sentence Transformer Name: {sentence_transformer_model}')
 
 embedder = SentenceTransformersDocumentEmbedder(
     model=sentence_transformer_model,
-    # model="T-Systems-onsite/german-roberta-sentence-transformer-v2",
-    # model="jinaai/jina-embeddings-v2-base-de",
-    # token=HUGGING_FACE_HUB_TOKEN
 )
-
-# hg_embedder = SentenceTransformer(
-#     "jinaai/jina-embeddings-v2-base-de",
-#     token=HUGGING_FACE_HUB_TOKEN
-# )
-
 embedder.warm_up()
 
-documents_with_embeddings = embedder.run(input_documents)
-# documents_with_embeddings = embedder.encode(input_documents)
 
+if EMBEDDING_CACHE_FILE and os.path.isfile(EMBEDDING_CACHE_FILE):
+    print("[INFO] Loading embeddings from cache")
 
-# print('\n\n')
-# # print(documents_with_embeddings['documents'])
-# print(type(documents_with_embeddings['documents']))
-# print(len(documents_with_embeddings['documents']))
-# print(dir(documents_with_embeddings['documents'][0]))
-# print('\n\n')
-# print(type(embedder.model))
-# print('\n\n')
-# # print(dir(hg_embedder))
+    with open(EMBEDDING_CACHE_FILE, 'r') as f:
+        documentsDict = json.load(f)
+        document_store.write_documents(
+            documents=[Document.from_dict(d) for d in documentsDict],
+            policy=DuplicatePolicy.OVERWRITE
+        )
 
+else:
+    print("[INFO] Generating embeddings")
 
-document_store.write_documents(
-    documents=documents_with_embeddings['documents'],
-    policy=DuplicatePolicy.OVERWRITE
-)
+    embedded = embedder.run(input_documents)
+    document_store.write_documents(
+        documents=embedded['documents'],
+        policy=DuplicatePolicy.OVERWRITE
+    )
 
-retriever = InMemoryEmbeddingRetriever(
-    # embedding_model="sentence-transformers/all-MiniLM-L6-v2",
-    document_store=document_store,
-    top_k=top_k
-)
+    if EMBEDDING_CACHE_FILE:
+        with open(EMBEDDING_CACHE_FILE, 'w') as f:
+            documentsDict = [Document.to_dict(d) for d in embedded['documents']]
+            json.dump(documentsDict, f)
```

Comment:
Following the [PEP8](https://peps.python.org/pep-0008/) standards, it is highly recommended to avoid single letter variable names and to use snake_case here:

```python
    if EMBEDDING_CACHE_FILE:
        with open(EMBEDDING_CACHE_FILE, 'w') as f_out:
            documents_dict = [Document.to_dict(d_) for d_ in embedded['documents']]
            json.dump(documents_dict, f_out)
```

Reply:
Updated in #24 by renaming f to f_out, documentsDict to documents_dict and d to d_. Exactly as seen above.
If you confirm, then we can close this review comment.